### PR TITLE
SKU Search Blogpost

### PR DESCRIPTION
**Pull request by khancoveo**
Submission for the Coveo tech blog.  The subject matter is searching by partial SKU/part ID in commerce and complex manufacturing.

### CODE REVIEW

File: _posts/2021-06-21-partial-sku-search.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+To get this code to access your fields, set `sku_meta` to your original or existing SKU field and `sku_field` to the new field you created to hold the list of partial values.  In our example, we used:
+
+{% highlight python %}
+sku_meta = 'bb_sku'
+sku_field = 'product_partial_match'
+{% endhighlight %}
+
+## Tying it together
+
+After you save the new extension, you must associate it to the indexing source. This is similar to the field mapping process: select the catalog source in the Administration Console, and then from the More menu item, select the Manage Extensions option.
```
Comment (francoislg) [inline _posts/2021-06-21-partial-sku-search.md:109]: Nitpick, put "More" and "Manage Extensions" in quotes for emphasis?

> then from the "More" menu item, select the "Manage Extensions" option.

File: _posts/2021-06-21-partial-sku-search.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+Select the extension you created earlier and then adjust the settings for applying the extension on the source items. For the Stage option, choose post-conversion. We recommend that you choose the ‘skip extension’ option for the error action and apply to all items since the reject option will eliminate the document from the index. These are the options highlighted in the screenshot below.
```
Comment (francoislg) [inline _posts/2021-06-21-partial-sku-search.md:111]: > For the "Stage" option

**Review by francoislg — APPROVED** [review]
Read through it, not much to suggest!

Really good read and the screenshots with the highlights are super nice! :smile:

**Comment by francoislg** [comment]
Here is the rendered link, by the way: https://khancoveo.github.io/source.coveo.com/2021/06/21/partial-sku-search/

Instead of using extensions, people could just push the data to the Push API pre-decomposed, right? Should it be mentioned that it could be done by whatever service is pushing the data as well? :sweat_smile:

File: _posts/2021-06-21-partial-sku-search.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+tags: [Complex manufacturing, eCommerce, SKU Search, B2B, Indexing Extensions]
+
+author:
+  name: Kate Hanratty
+  bio: Technical Enablement Manager, Education Services
+  twitter: 
+  image: khanratty.jpg
+
+---
+Do you want to have better and faster search results in your Coveo-powered catalog search pages?  You can do it by creating an indexing pipeline extension ([IPE](https://docs.coveo.com/en/3394/index-content/use-indexing-pipeline-extensions)) that identifies and stores all the variations of your partial SKU values.
```
Comment (amoreauCoveo) [inline _posts/2021-06-21-partial-sku-search.md:16]: So Jekyll already realizes that double spaces should be single spaces. So thankfully, nothing to report on that front :P

File: _posts/2021-06-21-partial-sku-search.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+Now, a retail or occasional customer might be willing to search by keyword and sift through the results, but commercial buyers are more likely looking for a specific product to order or reorder. And, they expect rapid retrieval from these large data sets, even if they don’t use the complete product ID, or SKU, when searching.
+
+While you could simply make the SKU field searchable with wildcards, you can make the response time up to 10 times faster by using SKU decomposition.
+
+## What is SKU Decomposition, anyway?
+
+Suppose one of your products has a SKU value of SJSH9-000-3K, but your buyers don’t always enter the full value. They might only enter SJSH9, for example, since it represents the category of product that they are looking for.  You can identify character sequences in the SKU and link them to the products containing the partial values.  For the example above, some variations of the SKU value are: S, SJ, SJS, SJSH, SHSJ9, 0, 00, 000, 3, 3K. This slicing and dicing of the SKU value is called decomposition.
```
Comment (amoreauCoveo) [inline _posts/2021-06-21-partial-sku-search.md:28]: SJSH9 (like my comment in Google doc :P )

File: _posts/2021-06-21-partial-sku-search.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+By working out the list in advance and storing it in the index, your Coveo-powered search interface will be able to find and surface relevant results without spending precious query time scanning the contents of all possible matching items.
+
+## Field Management
+
+If your catalog data is well structured and the SKU field already exists, then all you need to do is create a new field and map it to the SKU field. You can do this from Coveo Platform Administration Console using the [Add field](https://docs.coveo.com/en/1982/index-content/add-or-edit-a-field) option from the Content: Fields menu. Make sure you select the 'string' type and check the 'Multi-Value Facet' and 'Free text search' checkboxes, as highlighted in the screenshot below. In this example, our new field is called product_partial_match.
```
Comment (amoreauCoveo) [inline _posts/2021-06-21-partial-sku-search.md:34]: from the Coveo Platform . . . ( missing the)

File: _posts/2021-06-21-partial-sku-search.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+![Adding a new field from the Coveo Administration Console]({{ site.baseurl }}/images/2021-06-21-partial-sku-search/Add_New_Field_650_rnd.png "Adding a new field from the Coveo Administration Console.")
+
+Then, you can map this field to your SKU metadata field for the catalog source.  In the screenshot below, the new field `product_partial_match` is associated with the existing field for the actual SKU value: `bb_sku`.
+![Add a field mapping]({{ site.baseurl }}/images/2021-06-21-partial-sku-search/Add_field_mapping_full_rnd.png "Adding a field mapping to link the new field to the SKU field.")
+
+Check out our documentation for more details on how to add a [mapping rule](https://docs.coveo.com/en/1640/index-content/manage-source-mappings#add-or-edit-a-mapping-rule).
+
+For sources that use the Push API you can take another approach. If you compute the list of partial SKU values and submit them as another field in the index, then no indexing extension is required.  Just make sure that your field has the characteristics described here: 'string' type, 'Multi-Value Facet' and 'Free text search'.
```
Comment (amoreauCoveo) [inline _posts/2021-06-21-partial-sku-search.md:42]: missing Oxford comma at the end ☹️

**Review by amoreauCoveo — APPROVED** [review]

File: _posts/2021-06-21-partial-sku-search.md (hunk 4 above)
Comment (khancoveo) [reply _posts/2021-06-21-partial-sku-search.md:28]: Fixed, finally!

File: _posts/2021-06-21-partial-sku-search.md (hunk 2 above)
Comment (khancoveo) [reply _posts/2021-06-21-partial-sku-search.md]: Done.

File: _posts/2021-06-21-partial-sku-search.md (hunk 1 above)
Comment (khancoveo) [reply _posts/2021-06-21-partial-sku-search.md]: Done.

File: _posts/2021-06-21-partial-sku-search.md (hunk 5 above)
Comment (khancoveo) [reply _posts/2021-06-21-partial-sku-search.md:34]: Corrected.

File: _posts/2021-06-21-partial-sku-search.md (hunk 6 above)
Comment (khancoveo) [reply _posts/2021-06-21-partial-sku-search.md:42]: Fixed.

File: _posts/2021-06-21-partial-sku-search.md (hunk 3 above)
Comment (khancoveo) [reply _posts/2021-06-21-partial-sku-search.md]: Shame and pride made me clean this up too. 😁

**Pull request merged**